src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,53 @@
-import React from 'react';
+import * as React from 'react';
 import './CurrentUser.scss';
+import { getUsers } from '../../api/api';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+interface Props {
+  selectedUserId: number,
+}
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+type State = {
+  user: User | '',
+};
+
+export class CurrentUser extends React.Component<Props, State> {
+  state: State = {
+    user: '',
+  };
+
+  async componentDidMount() {
+    const newUser = await getUsers(this.props.selectedUserId);
+
+    // eslint-disable-next-line react/no-did-update-set-state
+    this.setState({ user: newUser });
+  }
+
+  async componentDidUpdate(prevProps: Props) {
+    if (prevProps.selectedUserId !== this.props.selectedUserId) {
+      const newUser = await getUsers(this.props.selectedUserId);
+
+      // eslint-disable-next-line react/no-did-update-set-state
+      this.setState({ user: newUser });
+    }
+  }
+
+  render() {
+    const { user } = this.state;
+
+    return (
+      <div className="CurrentUser">
+        <h2 className="CurrentUser__title">
+          <span>
+            Selected user:
+            {' '}
+            {user !== '' ? user.id : user}
```

Comment:
make check for an all block (after return). So you no need to check it on every place where you need to use user